sklearn/neighbors/_nearest_centroid.py

```diff
@@ -154,16 +154,16 @@ def fit(self, X, y):
                 self.centroids_[cur_class] = X[center_mask].mean(axis=0)
 
         if self.shrink_threshold:
+            if np.ptp(X).sum() == 0:
```

Comment:
I suspect we still need the `axis=0` to do ptp for each feature:

```suggestion
            if np.all(np.ptp(X, axis=0) == 0):
```

Also avoiding the `sum` avoids more numerical instability.

Reply:
Apologies, pushed before I noticed your review...will change to np.all methodology.